onnxruntime/core/optimizer/transformer_memcpy.cc

```diff
@@ -76,7 +76,8 @@ common::Status MemcpyTransformer::ApplyImpl(Graph& graph, bool& modified, int gr
         provider != onnxruntime::kVitisAIExecutionProvider &&
         provider != onnxruntime::kOpenVINOExecutionProvider &&
         provider != onnxruntime::kNnapiExecutionProvider &&
-        provider != onnxruntime::kAclExecutionProvider) {
+        provider != onnxruntime::kAclExecutionProvider &&
+        provider != onnxruntime::kArmNNExecutionProvider) {
```

Comment:
Probably need to update utils::ProviderIsCpuBased in onnxruntime/core/framework/utils.cc